### DISCUSSION TITLE
Foundry project yarn add hardhat

### DISCUSSION THREAD

**Discussion by DCVglobalnetwork** [Cyfrin/foundry-full-course-cu#1705] [General]
Hi everyone 
Anyone knows what is the best solution ?

I was trying to add hardhat 

```shell
yarn add Hardhat
```

![image](https://github.com/Cyfrin/foundry-full-course-f23/assets/105791829/0e1f2cfd-8d88-49d6-ac5e-5f8e8a26b99e)

me installing Node on Ubuntu 

![image](https://github.com/Cyfrin/foundry-full-course-f23/assets/105791829/388f6cb9-6aeb-4d7d-9d70-262ed7e8b315)

![image](https://github.com/Cyfrin/foundry-full-course-f23/assets/105791829/3361032d-65c8-4263-92dc-ae88e99b95d0)

????

![image](https://github.com/Cyfrin/foundry-full-course-f23/assets/105791829/3c7d49a9-ad18-458b-ad98-62e24f67bc28)

![image](https://github.com/Cyfrin/foundry-full-course-f23/assets/105791829/b22fbc3a-6d87-4bbd-a274-b31cb80a2f9c)

**Comment by EngrPips** [comment]
@DCVglobalnetwork, Did you resolve this already?

**Reply by DCVglobalnetwork** [reply]
Not yet

**Reply by EngrPips** [reply]
It says that Nodejs 4 or higher must be installed before you can add Hardhat. Have you installed the nodejs?

**Reply by DCVglobalnetwork** [reply]
yes I know it says Nodejs 4 , can you send me a link where I can get Nodejs 4 ? I have node.js on windows and Ubuntu node v12.22.9 that why I am confused

**Reply by EngrPips** [reply]
Then it is supposed to work as it requires `nodejs 4` or above, not specifically `nodejs 4`. Maybe run `node --version` in that same terminal where you ran `yarn add hardhat` and let's see what you will get.
